BUG: Fix f2py compile function testing.

charris opened the pull request:
- Create new file `test_compile_function.py` for the test, transferring
  the content from `test_quoted_characters.py` where it was before. The
  tricky part here is maintaining the history in both files after the
  move.
    
- The tests were failing on platforms without a fortran compiler as that
  was not being checked for. Add that check among others and clean up the
  test functions a bit.



<!-- Please be sure you are following the instructions in the dev guidelines
http://www.numpy.org/devdocs/dev/gitwash/development_workflow.html
-->

<!-- We'd appreciate it if your commit message is properly formatted
http://www.numpy.org/devdocs/dev/gitwash/development_workflow.html#writing-the-commit-message
-->

charris committed:
TST, MAINT: Separate f2py compile function testing into own file

Create new file `test_compile_function.py` for the test, transferring
the content from `test_quoted_characters.py` where it was before. The
tricky part here is maintaining the history in both files after the
move.

tylerjereddy reviewed (approved):
CI is all green -- looks like the straight migration of my misplaced unit tests as previously discussed along with helpful URLs to the corresponding PRs to alleviate confusion.

eric-wieser commented on numpy/f2py/tests/test_compile_function.py:
```diff
@@ -0,0 +1,121 @@
+"""See https://github.com/numpy/numpy/pull/11937.
```
I'm not sure this reference is useful

charris replied:
I sort of went back and forth, but on looking into what some of the other modules tested it simply seemed easier to link to the PR than trying to explain it ;) The whole series could use better documentation but it will be a dirty job.

charris committed:
BUG: Fix f2py test_compile_function.py tests.

The tests were failing on platforms without a fortran compiler as that
was not being checked for. Add that check among others and clean up the
test functions a bit.

charris commented:
Updated. This should fix the failing numpy-wheels nightly builds, but we won't know until it is merged.

charris commented:
I'm going to put this in to see what is does for numpy-wheels.
